tests/aws/services/s3/test_s3.py

```diff
@@ -7569,6 +7569,7 @@ def test_presigned_url_v4_x_amz_in_qs(
         head_object = aws_client.s3.head_object(Bucket=function_name, Key=object_key)
         snapshot.match("head-object", head_object)
 
+    @pytest.mark.skip(reason="Current test needs to be rewritten to work with AWS SDK v3")
```

Comment:
Temporary added skip so I can more easily focus on other failing tests in pipeline.
This skip needs to be removed before merge.

Reply:
Hi @bentsku 
Could you take a look at this test?
We need to update runtime to nodejs20 as node16 is getting deprecated and nodejs20 supports aws sdk v3.
I tried updating it, but sadly my JavaScript skills are lacking.

I think this should be of help
https://github.com/aws/aws-sdk-js-v3/issues/1576

Thank you

Reply:
Hmm, this is a bit tricky, because I believe this test is actively testing against the AWS JS SDK v2, if we use v3 then the test is not useful anymore. 

We would need to install the AWS JS SDK v2 inside the lambda in order for the test to stay relevant. Worst case, you can leave it skipped and I'll tackle it in a follow up. But the lambda using the AWS JS SDK v2 should stay as is for now.

Reply:
Thanks for giving it a look.


We are introducing a flag (off by default until next major version, I think). 
Switching it on would prevent starting lambda with this runtime and crash the test, but in theory default CI runs should run without issues until then.

I'm not sure if it would be better to leave test as it is, or mark it as skip?
What do you think?